pallets/statistics/src/lib.rs

```diff
@@ -186,12 +187,8 @@ impl<T: Config> Module<T> {
     ) -> DispatchResult {
         // Check EA permissions for asset.
         let did = Self::ensure_asset_perms(origin, asset)?;
-
-        // Check StatType per Asset limit.
-        ensure!(
-            stat_types.len() < T::MaxStatsPerAsset::get() as usize,
-            Error::<T>::StatTypeLimitReached
-        );
+        // converting from a btreeset to a bounded version 
+        let stat_type: BoundedBTreeSet<_, T::MaxStatsPerAsset> = stat_types.try_into().map(|_| Err(Error::<T>::StatTypeLimitReached))?;
```

Comment:
Need to use `map_err` to change the error type, `map` is used to change the `Ok` type.
```suggestion
        let stat_type: BoundedBTreeSet<_, T::MaxStatsPerAsset> = stat_types.try_into().map_err(|_| Error::<T>::StatTypeLimitReached)?;
```